api/v1alpha1/driver_types.go

```diff
@@ -28,9 +28,17 @@ type LogSpec struct {
 	// Log level for driver pods,
 	// Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.
 	// Default to 0
-	LogLevel   int               `json:"logLevel,omitempty"`
-	MaxFiles   int               `json:"maxFiles,omitempty"`
+	LogLevel int `json:"logLevel,omitempty"`
+	// MaxFiles is the number of logrtoate files
+	MaxFiles int `json:"maxFiles,omitempty"`
+	// MaxLogSize is the maximum size of the log per csi pods.
 	MaxLogSize resource.Quantity `json:"maxLogSize,omitempty"`
+	// Periodicity is the periodicity of the log rotation.
+	// +kubebuilder:validation:Pattern=`^$|^(hourly|daily|weekly|monthly|1h|24h|1d)$`
+	// +optional
+	Periodicity string `json:"periodicity,omitempty"`
```

Comment:
If this is supposed to be an enum with specific values, please substitute the `string` type with an `alias string` type backed by constants.

Reply:
Updating

Reply:
@parth-gr hourly and 1hour are both same right? if yes, can you please remove 1h from here we already have hourly,

Reply:
Done

Reply:
>specialization of a string.

Can you provide an example usage, I don't understand what does this mean

Reply:
@nb-ohad ^^

Reply:
@parth-gr Please take a look at this as an example: 
https://github.com/ceph/ceph-csi-operator/blob/main/api/v1alpha1/driver_types.go#L36-L51

Reply:
Thanks done